pkg/providers/equinix.go

```diff
@@ -79,22 +79,26 @@ func (p *equinixProvider) Acquire(poolSize int, poolName string, poolType string
 	uniqueId := strings.Replace(uuid.New().String(), "-", "", -1)
 	resourceName := fmt.Sprintf("%s-%s", hostPrefix, uniqueId)
 
-	cr := packngo.DeviceCreateRequest{
-		Hostname:  resourceName,
-		Metro:     p.config.Metro,
-		Plan:      p.config.Plan,
-		OS:        p.config.OS,
-		ProjectID: p.config.ProjectID,
-		Tags:      []string{poolName},
-	}
+	for _, metro := range p.config.Metros {
+		cr := packngo.DeviceCreateRequest{
+			Hostname:  resourceName,
+			Metro:     metro,
+			Plan:      p.config.Plan,
+			OS:        p.config.OS,
+			ProjectID: p.config.ProjectID,
+			Tags:      []string{poolName},
+		}
 
-	device, _, err := p.client.Devices.Create(&cr)
-	if err != nil {
-		return resource, fmt.Errorf("error creating device: %w", err)
+		device, _, err := p.client.Devices.Create(&cr)
+		if err != nil {
+			continue
```

Comment:
It would be useful to have a log trace here describing which was the error that forced us to skip the current metro

Reply:
The logger is configured in the cir fsm. I don't know how to bring it inside the provider.

Reply:
A `logger` instance could be simply passed through the `providers.NewProvider` (invoked just in `CIResourceFSM.Process` and then stored in the selected factory